### DISCUSSION TITLE
Multitenancy SuperAdmin Portal

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#3071] [Feature Requests]
A layer above Directus that allows seamless management of multiple projects:

* Seeing all available options
* Users straddling projects
* Switching between projects
* Staying authenticated (Directus SSO?)
* Shared extensions?

**Comment by salarelv** [comment]
If it was me I would separate the user database into a separate database. So they could be global for each project. And have a list of projects that the user has access to.
How to run multiple projects with the same nodejs instance? I have a need for small websites that I can customize but I do not want to keep running multiple instances on directus. These projects could run on sqlite fine.

**Reply by benhaynes** [reply]
True... or we could look into an SSO option for auth across instances. It'd be nice if we could avoid having a new process running for the superadmin side, perhaps there's a way to keep it in Directus Core itself, so we mesh all the instances together. 🤔

**Reply by salarelv** [reply]
I am working in a company that has a PHP backend with multi-tenacy. Idea is similar as I described. Users and other global stuff is separated into a separate database and there are project database'is. Project Id is passed into every API call (like a token). System checks if this project exists and opens a database connection to this database.

**Reply by gearoidc** [reply]
> How to run multiple projects with the same nodejs instance? I have a need for small websites that I can customize but I do not want to keep running multiple instances on directus. These projects could run on sqlite fine.

Im in the same boat too. Would be cool to give some of my clients access to the same Directus setup where they can manage content for each of their own sites, rather than having to setup many instances of Directus. My apps then would have an API key p/site.

**Comment by gisostallenberg** [comment]
I'm looking for a feature where at least the tenant (a separate database) is selected per domain that points to the Directus runtime.
It could be by providing a mapping with domains and a database, e.g.
```json
{
    "tenant-a.example.com": "TenantA_database",
    "tenant-A.example.com": "TenantA_database",
    "tenant-b.example.com": "MyTenant_database_for_customerB"
}
```

Off course it would be lovely if this can be maintained with some admin interface (without a json), but just being able to select separate databases this way would be a huge plus for me.

**Reply by xu4wang** [reply]
odoo is an ERP system with similar multi-tenant design. 

[Odoo is a multi-tenant system](https://www.odoo.com/documentation/15.0/administration/install/deploy.html#dbfilter)

In the configuration file, the db name is not a string, but can be a regex. For example,
     dbfilter = ^%d$

For any incoming API request, the system will check the request URL, and select the first subdomain after www as the database name.

For example, the node process is listening on port 8055, both tenant-a.example.com and tenant-b.example.com are forwarded to that port.  

Below two API calls will be processed by the same node instance listening on port 8055.
- http://tenant-a.example.com/items/Products
- http://tenant-b.example.com/items/Products

However, the first request will be processed with database tenant-a, and the second one will be with tenant-b.

The benefit is that only one nodejs instance is required for all the tenants.

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
